hudi-common/src/main/avro/HoodieArchivedMetaEntry.avsc

```diff
@@ -128,6 +128,11 @@
             "HoodieIndexCommitMetadata"
          ],
          "default": null
+      },
+      {
+         "name":"stateTransitionTime",
+         "type":["null","string"],
+         "default": null
```

Comment:
This could cause backward compatibility issue, if the table is upgraded, this field could be missing from the old table, we need to address this in release note at least.

Reply:
its a new filed and default is null. so, it is backwards compatible. But curious to know why do we need this in archival entry. what benefit we get by adding this entry to archived entry? can we just ignore it.

Reply:
Saving stateTransition time in archiver can help us to do some troubleshoot work relate to timeline(if we drop modification time after the instant is archived, we can not identify how downstream jobs ingesting these instants).

Actually in our internal version, we change the `Call show commits/ archive commits` to also show stateTransition time. If this implementation is acceptable, I'll add that here as well.

Reply:
+1 we should save it in the archived metadata. I can see other potential use cases when there can be holes in the timeline after we allow archival beyond savepoint.

Reply:
> Actually in our internal version, we change the Call show commits/ archive commits to also show stateTransition time. If this implementation is acceptable, I'll add that here as well.

@boneanxs It's always good to have CLI updated accordingly.